R/animint.R

```diff
@@ -649,12 +694,26 @@ saveLayer <- function(l, d, meta){
 saveChunks <- function(x, vars, meta){
   if(is.data.frame(x)){
     if(length(vars) == 0){
+      if(meta$geom %in% c("line", "path", "polygon", "ribbon")) {
+        if("group" %in% names(x)) {
+          # common info for groups, save only once
+          csv.name <- sprintf("%s_chunk_shape.tsv", meta$classed)
+          chunk.shape <- file.path(meta$out.dir, csv.name)
+          if(!file.exists(chunk.shape)){
+            x1 <- x[c("x", "y", "group")]
+            write.table(x1, chunk.shape, quote = FALSE, row.names = FALSE, 
+                        sep = "\t")
+          }
+          # info for each group
+          x2 <- x[, !names(x) %in% c("x", "y")]
+          x <- x2[!duplicated(x2), ]
+          }
```

Comment:
Can you please use more informative variable names than `x1`, `x2`?

For example, `chunk.specific` instead of `x2` and `same.in.all.chunks` for `x1`.

Also, do you think it is necessary to check that the geom is a line/path/polygon/ribbon? I think this is OK for a first attempt, but eventually we should use a more relevant test: does this geom have any repeated data? (maybe not necessarily just x,y columns)